BASE/src/Microsoft.ApplicationInsights/ActivityExtensions.cs

```diff
@@ -47,29 +50,62 @@ internal static void SetOperationName(this Activity activity, string operationNa
+        private static bool IsActivityTypeAvailable()
         {
-            try
+            if (activityTypeAvailability == ActivityAvailable)
+            {
+                // Fast common case
+                return true;
+            }
+
+            if (activityTypeAvailability == ActivityNotAvailable)
+            {
+                // This can happen if System.Diagnostics.DiagnosticSource cannot be loaded,
+                // i.e. in some PowerShell scenarios where only Microsoft.ApplicationInsights.dll was loaded.
+                return false;
+            }
+
+            // Must be activityTypeAvailability == ActivityAvailabilityUnknown
+
+            if (CanLoadActivityType())
             {
-                Assembly.Load(new AssemblyName("System.Diagnostics.DiagnosticSource, Version=4.0.4.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51"));
+                Interlocked.Exchange(ref activityTypeAvailability, ActivityAvailable);
                 return true;
             }
-            catch (System.IO.FileNotFoundException)
+            else
             {
-                // This is a workaround that allows ApplicationInsights Core SDK run without DiagnosticSource.dll
-                // so the ApplicationInsights.dll could be used alone to track telemetry, and will fall back to CallContext/AsyncLocal instead of Activity
+                Interlocked.Exchange(ref activityTypeAvailability, ActivityNotAvailable);
                 return false;
             }
-            catch (System.IO.FileLoadException)
+        }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        private static bool CanLoadActivityType()
+        {
+            // This is a workaround that allows ApplicationInsights Core SDK run without DiagnosticSource.dll
+            // so the ApplicationInsights.dll could be used alone to track telemetry, and will fall back to CallContext/AsyncLocal instead of Activity
+            try
+            {
+                return LoadActivityType();
+            }
+            catch (System.IO.IOException)
             {
-                // Dll version, public token or culture is different
+                // Cannot open DLL file
                 return false;
             }
-            finally
+            catch (System.TypeLoadException)
             {
-                isInitialized = true;
+                // Cannot resolve type
+                return false;
             }
         }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        private static bool LoadActivityType()
+        {
+            // At this point the Activity type has already been loaded, or - if not - the runtime will attempt to load the assembly from which it is referenced.
+            // While doing so, it will respect type redirects potentially set up by the execution environment. This is preferable to doing Assembly-Load directly.
+            return typeof(System.Diagnostics.Activity) != null;
```

Comment:
SDK must have a particular version of the assembly, as it relies of features not available in older versions. (like w3c format etc.) This was probably the motivation behind original code trying to load the exact same version.
With this code, we may break else where, if an Activity of older version is available, and we attempt to call methods not available in that version.